src/thriveopedia/pages/ThriveopediaEvolutionaryTreePage.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///    Thriveopedia page displaying an evolutionary tree of all species in the current game.
+/// </summary>
+/// <remarks>
+///   <para>
+///     Note a lot of this functionality is duplicated from AutoEvoExploringTool.
+///   </para>
+/// </remarks>
+public class ThriveopediaEvolutionaryTreePage : ThriveopediaPage
+{
+    [Export]
+    public NodePath DisabledInFreebuildPath = null!;
+
+    [Export]
+    public NodePath EvolutionaryTreePath = null!;
+
+    [Export]
+    public NodePath SpeciesDetailsLabelPath = null!;
+
+    [Export]
+    public NodePath SpeciesPreviewPath = null!;
+
+    [Export]
+    public NodePath HexPreviewPath = null!;
+
+    private readonly List<Dictionary<uint, Species>> speciesHistoryList = new();
+    private VBoxContainer disabledInFreebuild = null!;
+    private EvolutionaryTree evolutionaryTree = null!;
+    private CustomRichTextLabel speciesDetailsLabel = null!;
+    private SpeciesPreview speciesPreview = null!;
+    private CellHexesPreview hexesPreview = null!;
+
+    public override string PageName => "EvolutionaryTree";
+    public override string TranslatedPageName => TranslationServer.Translate("EVOLUTIONARY_TREE_PAGE");
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        disabledInFreebuild = GetNode<VBoxContainer>(DisabledInFreebuildPath);
+        evolutionaryTree = GetNode<EvolutionaryTree>(EvolutionaryTreePath);
+        speciesDetailsLabel = GetNode<CustomRichTextLabel>(SpeciesDetailsLabelPath);
+        speciesPreview = GetNode<SpeciesPreview>(SpeciesPreviewPath);
+        hexesPreview = GetNode<CellHexesPreview>(HexPreviewPath);
+
+        UpdateCurrentWorldDetails();
+    }
+
+    public override void UpdateCurrentWorldDetails()
+    {
+        if (CurrentGame == null)
+            return;
+
+        RebuildTree();
+    }
+
+    public override void OnNavigationPanelSizeChanged(bool collapsed)
+    {
+    }
+
+    /// <summary>
+    ///   Clear and then rebuild the evolutionary tree each time we open the page. This way, we ensure the tree is
+    ///   always up to date.
+    /// </summary>
+    public void RebuildTree()
+    {
+        if (!Visible)
+            return;
+
+        // TODO fix the tree for freebuild?
+        if (CurrentGame!.FreeBuild)
+        {
+            evolutionaryTree.Visible = false;
+            disabledInFreebuild.Visible = true;
+            return;
+        }
+
+        // Building the tree relies on the existence of a full history of generations stored in the current game. Since
+        // we only started adding these in 0.6.0, it's impossible to build a tree in older saves.
+        // TODO avoid an ugly try/catch block by actually checking the original save version?
```

Comment:
Doing the TODO here would be a pretty good idea

Reply:
Is there a good way to go about getting this information? I had a look into this but found no leads.

Reply:
Well we would actually need a new field in the save info to preserve this info... so kind of probably not worth it (but overall wouldn't be too hard to add, but kind of the point of the save upgrade system is that old saves work as well as new ones). Maybe it'd be just better then to add proper error printing and showing right info to the player (like my next comment suggests).